org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java

```diff
@@ -100,37 +106,47 @@ private void importDir(Path rootFolder, IProgressMonitor monitor) {
 				}
 			}
 		}
-		startSynchronization(rootFolder.toFile(), distribution, NewProjectHandler.IMPORT_AND_MERGE);
+		return distribution;
 	}
 
-	protected void startSynchronization(File location, GradleDistribution distribution, NewProjectHandler newProjectHandler) {
-		List<String> jvmArgs = new ArrayList<>();
-		IProxyService proxyService = JavaLanguageServerPlugin.getInstance().getProxyService();
-		IProxyData proxy = proxyService.getProxyData(IProxyData.HTTP_PROXY_TYPE);
-		if (proxy != null) {
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTP_PROXY_HOST, proxy.getHost());
-			if (proxy.getPort() > 0) {
-				addArgs(jvmArgs, JavaLanguageServerPlugin.HTTP_PROXY_PORT, String.valueOf(proxy.getPort()));
-			}
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTP_PROXY_USER, proxy.getUserId());
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTP_PROXY_PASSWORD, proxy.getPassword());
-		}
-		proxy = proxyService.getProxyData(IProxyData.HTTPS_PROXY_TYPE);
-		if (proxy != null) {
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTPS_PROXY_HOST, proxy.getHost());
-			if (proxy.getPort() > 0) {
-				addArgs(jvmArgs, JavaLanguageServerPlugin.HTTPS_PROXY_PORT, String.valueOf(proxy.getPort()));
+	protected void startSynchronization(Path rootFolder, IProgressMonitor monitor) {
+		boolean shouldSynchronize = true;
+		File location = rootFolder.toFile();
+		IProject[] projects = ResourcesPlugin.getWorkspace().getRoot().getProjects();
+		for (IProject project : projects) {
+			if (project.exists() && ProjectUtils.isGradleProject(project)) {
+				File projectDir = project.getLocation() == null ? null : project.getLocation().toFile();
+				if (location.equals(projectDir)) {
+					PersistentModel model = CorePlugin.modelPersistence().loadModel(project);
+					if (model.isPresent()) {
+						File persistentFile = CorePlugin.getInstance().getStateLocation().append("project-preferences").append(project.getName()).toFile();
```

Comment:
the whole "check if preferences are stale" block should be extracted to another method, that can be unit tested